### ISSUE TITLE
Input: clear icon is overflowing focus outline when in hover or active state

### ISSUE THREAD

**Issue by Lukas742** [SAP/ui5-webcomponents#4839]
### **Bug Description**
When an `Input` is focused and the clear button is defined by setting `showClearIcon`, the button is overflowing the outline of the input when in hover or active state.

### **Expected Behavior**
The outline should overflow the button.

### **Steps to Reproduce**
1. Go to [playground (master)](https://sap.github.io/ui5-webcomponents/master/playground/components/Input/)
2. Focus the input with a clear icon.
3. Hover over the clear icon/button. --> See that the outline is covered by the clear button

### **Isolated Example**
https://sap.github.io/ui5-webcomponents/master/playground/components/Input/

<img width="237" alt="image" src="https://user-images.githubusercontent.com/9749730/156195513-45e445ee-51e1-43e5-a4e0-2af21ef7898f.png">

### **Context**
- UI5 Web Components version: 1.2.0

### **Priority**
- [x] Low
- [ ] Medium
- [ ] High
- [ ] Very High

**Comment by ilhan007** [comment]
Hello @SAP/ui5-webcomponents-topic-rl could you check this visual issue?

**Comment by Lukas742** [comment]
I just noticed that it differs with UI5 as well:

- Clear button is overflowing the outline for Input: https://openui5nightly.hana.ondemand.com/entity/sap.m.Input/sample/sap.m.sample.InputDescription

- Clear button is behind the outline for Searchfield:
https://openui5nightly.hana.ondemand.com/entity/sap.m.SearchField/sample/sap.m.sample.SearchField

I'm not sure what the expected specs here are, personally I like the 2nd one better.

**Comment by elenastoyanovaa** [comment]
Hello colleagues,

I have checked the issue in both the components and the controls. For sap.m.Input(Base) and ui5-input, the focus is set via outline and outline-offset styles. In the sap.m.SearchField the focus is applied with a ::before pseudo element. This is why the icon background on hover does not go over the focus outline of the control. 

What I noticed additionally. I have checked the visual spec for fiori 3 and inputs with icons. If you open the specification of the MultiInput and go to "Active Field" you will see a strange screenshot, showing a focus outline and an active icon, but the focus outline goes only around the input field part before the icon, not on the whole control. I think that this was never implemented as so in none of the input controls, so we should double check with the VD designers, before doing styling changes.
